### ISSUE TITLE
Warning: NaN or Inf found in input tensor.

### ISSUE THREAD

**Issue by benedictflorance** [VisionLearningGroup/DA_Detection#11]
I am getting this warning when I train from Cityscapes to Foggy Cityscapes. The RPN regression box loss becomes NaN.

I tried changing all these, but the error still goes NaN.

From https://github.com/jwyang/faster-rcnn.pytorch/issues/136#issuecomment-390544655
```
            x1 = max(float(bbox.find('xmin').text) - 1, 0)
            y1 = max(float(bbox.find('ymin').text) - 1, 0)
            x2 = max(float(bbox.find('xmax').text) - 1, 0)
            y2 = max(float(bbox.find('ymax').text) - 1, 0)
```

From https://github.com/jwyang/faster-rcnn.pytorch/issues/193#issuecomment-408690085
`not_keep = (gt_boxes[:,2] - gt_boxes[:,0]) < 10 and (gt_boxes[:,3] - gt_boxes[:,1]) < 10
`

It stills throws the warning. What may be the possible reason for the issue?

**Comment by orzyt** [comment]
Hey @benedictflorance , I have the same issue as you.

I think the ground truth in Cityscapes dataset may be inaccurate.
When the bounding box is flipped (since `cfg.TRAIN.USE_FLIPPED` is `True`), the coordinates may be negative.

So, I simply replace the following code

https://github.com/VisionLearningGroup/DA_Detection/blob/730eaca8528d22ed3aa6b4dbc1965828a697cf9a/lib/datasets/imdb.py#L122-L123

with  https://github.com/tiancity-NJU/da-faster-rcnn-PyTorch/blob/master/lib/datasets/imdb.py#L129-L131 .

Then, the RPN regression box loss is back to normal.

Hope this helps you. :smile:

**Comment by AmingWu** [comment]
@orzyt, I use your method to train the model. But I find the test result is 0.

**Comment by AmingWu** [comment]
Hey @benedictflorance，have you solved this problem?

**Comment by benedictflorance** [comment]
@AmingWu Sorry for the delayed response.

Adding this after the line `boxes[:, 2] = widths[i] - oldx1 - 1` in imdb.py worked.
```
      for b in range(len(boxes)):     
                if boxes[b][2] < boxes[b][0]:
                   boxes[b][0] = 0
```
